cmd/federation-controller/main.go

```diff
@@ -212,6 +212,7 @@ func main() {
 			go func() {
 				if err := federationClient.Run(); err != nil {
 					log.Fatalf("failed to start FDS client: %v", err)
+					time.AfterFunc(5*time.Second, federationClient.Restart)
```

Comment:
I think we have overlooked a side-effect of calling `log.Fatalf`. It will call `os.Exit(1)` deep under the hood. 

Considering the scope of this PR, we should change the log level here to `Errorf` to restart to happen.

However, I would suggest disallowing using `log.Fatalf` due to this hidden side effect and enforcing handling errors explicitly across the code base (in this case, calling os.Exit). The latter approach also gives us an opportunity to use exit statuses other than generic `1`, for example, `128`, which indicates incorrect arguments passed to the command. I can open an issue if we agree on that.

Reply:
Normally I would give you a code `suggestion` to apply, but it affects an unedited line. Not possible on GH (yet?).

Reply:
> I think we have overlooked a side-effect of calling log.Fatalf. It will call os.Exit(1) deep under the hood.

I was aware of the behavior of `log.Fatalf` - I just forgot to remove it in this PR, even though that was the main goal... Initially, I decided to use `Fatalf` to exit and trigger restart of the pod.